pkg/k8s/apis/cilium.io/v2/validator/validator.go

```diff
@@ -105,10 +109,89 @@ func (n *NPValidator) ValidateCNP(cnp *unstructured.Unstructured) error {
 	return nil
 }
 
+const (
+	// We can remove the check for this warning once 1.9 is the oldest supported Cilium version.
+	warnEmptyToFromEndpointMessage = "It seems you have a CiliumClusterwideNetworkPolicy " +
+		"with an empty to/from endpoint selector. The behavior of this selector has been " +
+		"changed. The selector now only allows traffic to/from Cilium managed K8s endpoints, " +
+		"instead of acting as a truly empty endpoint selector allowing all traffic. To " +
+		"ensure that the policy behavior does not affect your workloads, consider adding " +
+		"another policy that allows traffic to/from world and cluster entities. For a more " +
+		"detailed discussion on the topic, see GitHub issue #12844"
+)
+
 // ValidateCCNP validates the given CCNP accordingly the CCNP validation schema.
 func (n *NPValidator) ValidateCCNP(ccnp *unstructured.Unstructured) error {
 	if errs := validation.ValidateCustomResource(nil, &ccnp, n.ccnpValidator); len(errs) > 0 {
 		return errs.ToAggregate()
 	}
+
+	logger := log.WithFields(logrus.Fields{
+		logfields.CiliumClusterwideNetworkPolicyName: ccnp.GetName(),
+	})
+
+	// At this point we have validated the custom resource with the new CRV.
+	// We can try converting it to the new CCNP type.
+	// This should not fail, so we are not returning any errors, just logging
+	// a warning.
+	ccnpBytes, err := ccnp.MarshalJSON()
+	if err != nil {
+		return err
+	}
+
+	resCCNP := cilium_v2.CiliumClusterwideNetworkPolicy{}
+	err = json.Unmarshal(ccnpBytes, &resCCNP)
+	if err != nil {
+		return err
+	}
+
+	// Print the warninig only once per CCNP.
+	if resCCNP.Spec != nil {
+		if containsEmptyToFromEndpoint(resCCNP.Spec) {
+			logger.Warning(warnEmptyToFromEndpointMessage)
+			return nil
+		}
+	}
+
+	if resCCNP.Specs != nil {
+		for _, rule := range resCCNP.Specs {
+			if containsEmptyToFromEndpoint(rule) {
+				logger.Warning(warnEmptyToFromEndpointMessage)
```

Comment:
Will this log the warning even if the user intended for it to only select endpoints and not other entities? That's how I at least thought it worked from the beginning :/ Notifying users who are expecting the new behaviour is just extra noise. Would this be more appropriate in a release note / upgrade guide note instead?

Reply:
I think if we are not planning to backport the pull request then it makes sense to remove this extra noise and have this information in the upgrade guide/release note. But if we will be backporting the Pull requests then I think we should not consider that users expect this behavior and have plumbed their policies accordingly. The warning does make sense in such cases. 
 What are your thoughts on this @aanm ?